libavfilter/vf_scale_qsv.c

```diff
@@ -534,9 +208,43 @@ static int qsvscale_config_props(AVFilterLink *outlink)
     outlink->w = w;
     outlink->h = h;
 
-    ret = init_scale_session(ctx, inlink->w, inlink->h, w, h);
-    if (ret < 0)
-        return ret;
+    if (inlink->format == AV_PIX_FMT_QSV) {
+        if (!inlink->hw_frames_ctx || !inlink->hw_frames_ctx->data)
+            return AVERROR(EINVAL);
+        else
+            in_format = ((AVHWFramesContext*)inlink->hw_frames_ctx->data)->sw_format;
+    } else
+        in_format = inlink->format;
+
+    if (s->format == AV_PIX_FMT_NONE)
+        s->format = in_format;
+
+    outlink->frame_rate = inlink->frame_rate;
+    outlink->time_base = av_inv_q(inlink->frame_rate);
+    param.out_sw_format = s->format;
+
+#if QSV_HAVE_SCALING_CONFIG
+    param.ext_buf                      = ext_buf;
+    memset(&s->scale_conf, 0, sizeof(mfxExtVPPScaling));
+    s->scale_conf.Header.BufferId      = MFX_EXTBUFF_VPP_SCALING;
+    s->scale_conf.Header.BufferSz      = sizeof(mfxExtVPPScaling);
+    s->scale_conf.ScalingMode          = s->mode;
+    param.ext_buf[param.num_ext_buf++] = (mfxExtBuffer*)&s->scale_conf;
+    av_log(ctx, AV_LOG_VERBOSE, "Scaling mode: %d\n", s->mode);
+#endif
+
+    if (inlink->w != outlink->w ||
+        inlink->h != outlink->h ||
+        in_format != s->format) {
+        ret = ff_qsvvpp_init(ctx, &param);
```

Comment:
Not related to this. But, we may need to add some passthrough skip mode. 
for example, the customer may need to add subtitles on decode output.

Reply:
I'm sorry I didn't catch your point. User may not use this filter if they want 'passthrough skip mode'

Reply:
How to draw something on decode output? We need some filter copy input to output.

Reply:
I think you should use overlay_qsv instead of scaling_qsv for your case.